src/test/java/com/dawn/jat/illuminati/post/service/PostServiceTest.java

```diff
@@ -114,45 +126,55 @@ void create() {
                 .thenReturn(postEntity);
         Mockito.when(postRepository.savePost(any(PostEntity.class)))
                 .thenReturn(postEntity);
-        assertThat(postService.create(postDto), is(postEntity));
+        Mockito.when(modelMapper.map(any(), any())).thenReturn(postDto);
+        assertThat(postService.createPost(postDto), is(postDto));
         Mockito.verify(postRepository, Mockito.times(1)).savePost(postEntity);
     }
 
     @Test
     void savePost_WithDto_SuccessfullySaveDto() {
-
-        PostEntity expectedPost = new PostEntity("how-to-apply-agile-methodology",
+        PostEntity expectedPost = new PostEntity(
+                "how-to-apply-agile-methodology",
                 "How to apply Agile methodology new",
                 "How to apply Agile methodology new",
                 "01/01/2020 new",
                 new ArrayList<>(Arrays.asList("System Design", "OOP")),
-                "Li Li new");
+                "Li Li new",
+                "");
         expectedPost.setContent("new content");
-        Mockito.when(postRepository.findById(postDto.getId()))
-                .thenReturn(Optional.of(postEntity));
+        String id = postDto.getId();
+        assertNotNull(id);
+
+        Mockito.when(postRepository.existsById(id))
+                .thenReturn(true);
+
         Mockito.when(postRepository.findById(postDto.getId()))
                 .thenReturn(Optional.of(postEntity));
 
-        postService.save(postDto);
+        Mockito.when(converter.convertPostDtoToEntity(eq(postDto), any(PostEntity.class)))
+                .thenReturn(postEntity);
 
+        Mockito.when(postRepository.savePost(postEntity)).thenReturn(postEntity);
 
+        postService.savePost(postDto);
+        assertNotNull(postDto);
         Mockito.verify(postRepository, Mockito.times(1)).findById(postDto.getId());
     }
 
     @Test
-    void savePost_WithDto_ThrowPostNotFoundException() {
-        PostEntity expectedPost = new PostEntity("how-to-apply-agile-methodology",
-                "How to apply Agile methodology new",
-                "How to apply Agile methodology new",
-                "01/01/2020 new",
-                new ArrayList<>(Arrays.asList("System Design", "OOP")),
-                "Li Li new");
-        expectedPost.setContent("new content");
-        Mockito.when(postRepository.findById(postDto.getId()))
-                .thenReturn(Optional.empty());
+    void savePost_WithDto_getIdNullThrowPostCanNotBeSavedException_invalidPost() {
+        PostDto postDto = new PostDto();
+        postDto.setId(null);
```

Comment:
should `new PostDto()` will create PostDto with id = null 🤔 
Listing-1:
https://www.javaworld.com/article/3040564/java-101-class-and-object-initialization-in-java.html